# Build yasm instead of using the one included with Chromium on Windows

--- a/third_party/libjpeg_turbo/libjpeg.gyp
+++ b/third_party/libjpeg_turbo/libjpeg.gyp
@@ -198,8 +198,11 @@
         # Mac, we always build our patched yasm and use it because of
         # <http://www.tortall.net/projects/yasm/ticket/236>.
         [ 'OS=="win"', {
+          'dependencies': [
+            '../yasm/yasm.gyp:yasm#host',
+          ],
           'variables': {
-            'yasm_path': '../yasm/binaries/win/yasm<(EXECUTABLE_SUFFIX)',
+            'yasm_path': '<(PRODUCT_DIR)/yasm<(EXECUTABLE_SUFFIX)',
             'conditions': [
               [ 'target_arch=="ia32"', {
                 'yasm_format': '-fwin32',
--- a/third_party/skia/gyp/libjpeg-turbo.gyp
+++ b/third_party/skia/gyp/libjpeg-turbo.gyp
@@ -20,13 +20,9 @@
     {
       'target_name': 'yasm-win',
       'type': 'executable',
-      'sources': [
-        '../third_party/externals/yasm/binaries/win/yasm.exe',
+      'dependencies': [
+        '../third_party/yasm/yasm.gyp:yasm#host',
       ],
-      'copies' : [{
-        'destination': '<(PRODUCT_DIR)',
-        'files': [ '../third_party/externals/yasm/binaries/win/yasm.exe' ],
-      }],
     },
     {
       'target_name': 'libjpeg-turbo',
